components/more_direct_channels/more_direct_channels.jsx

```diff
@@ -177,23 +179,21 @@ export default class MoreDirectChannels extends React.Component {
 
         this.setState({saving: true});
 
-        const success = (channel) => {
-            // Due to how react-overlays Modal handles focus, we delay pushing
-            // the new channel information until the modal is fully exited.
-            // The channel information will be pushed in `handleExit`
-            this.exitToChannel = '/' + this.props.currentTeamName + '/channels/' + channel.name;
+        const done = (result) => {
+            const {data, error} = result;
             this.setState({saving: false});
-            this.handleHide();
-        };
 
-        const error = () => {
-            this.setState({saving: false});
+            if (!error) {
+                this.exitToChannel = '/' + this.props.currentTeamName + '/channels/' + data.name;
+                this.setState({saving: false});
```

Comment:
I think this line is not needed, the `saving` was set to false in the line 184.

Reply:
nice catch @jespino I'll remove it